src/main.rs

```diff
@@ -141,6 +179,11 @@ fn rocket() -> _ {
         panic!("Failure to parse configuration")
     });
 
+    base = match config.oauth_provider() {
+        OauthProvider::Google => base.attach(fairing_google()),
+        OauthProvider::Microsoft => base.attach(fairing_microsoft()),
+    };
```

Comment:
This seems like something we would need in every communication plugin, and that further would change whenever the set of authentication providers changes, so I would suggest we shift at least the match portion of this into comm-common, to only have one place where actual code changes are needed when adding new oauth providers.